### DISCUSSION TITLE
Taylor Green Vortex - 2D with fixed time step

### DISCUSSION THREAD

**Discussion by fab6** [exadg/exadg#222] [General] [answered]
Hi,
I would like to adjust as a starting point the taylor green vortex a bit. As a first step I would like to compare it with a simple openfoam inviscid setup ( I set the flag to inviscid).

For this I would like to run it in 2D, which works with the json file setting. Further I need to set the initial fields for U and p accordingly.

Another, adjustment what I would like to test is to set a fixed time step. I was thinking that adjusting this
![image](https://user-images.githubusercontent.com/3047695/162562764-f211e7ff-dcd0-494a-b6cb-30cfcbf791e1.png)

would be enough. But it does not work as hoped:
![image](https://user-images.githubusercontent.com/3047695/162562796-8c6d52d3-c8c8-4297-a88f-c40c85b83dff.png)

Do you have an advice how to fix this?
Thanks in advance!
Fab

**Comment by fab6** [comment]
Ok, I am one little step ahead, I just figured out that the `Makefile` is not even taking my adjusted `application.h` file into account. It still points to the initial directory.

**Comment by fab6** [comment]
Though I am having trouble to compile the example outside the build directory. When just copying one of the example applications to another directory, I have trouble to configure it with `cmake .`

I get following error (here it is the poiseuille example:
![image](https://user-images.githubusercontent.com/3047695/162606480-c84e9964-7014-431c-8a1b-9e7eda31bcd0.png)

For the deal.ii examples this works fine, e.g.:
![image](https://user-images.githubusercontent.com/3047695/162606594-b0199770-92f3-447e-a87a-222bb4800db2.png)


Do you have a short hint, how to compile it outside the build directories?
Thank you in advance!
Fab

**Reply by nfehn** [reply]
The design in exadg is to simply copy one example to another directory. Configure + make is then called in the exadg/build directory. I expect that other workflows require more adjustments regarding cmake. It should be possible to set up your own project and link against exadg, but this is probably way more work to achieve what you currently need.

**Reply by fab6** [reply]
Hm, excuse me, but I do not understand yet...
I have the initial `exadg` pulled directory with `exadg/applications` and after the inital build I have `exadg/build/applications`.

As an example for the `cavity` case it looks like:
- for the initial build directory:
![image](https://user-images.githubusercontent.com/3047695/162617246-020a76cc-4109-4a24-b0e2-3b764470f6fc.png)


- and for the non-build directory:
![image](https://user-images.githubusercontent.com/3047695/162617233-feacdca8-ff89-48f9-a378-a7f347bcfd46.png)


What works now is that I adjust the `application.h` in the non-build directory and get via `make` the `solver` in the build directory.  But I think this is not what you mean, is this right?
Thank you for your help!

**Comment by nfehn** [comment]
What exactly does nor work as expected regarding time stepping?

**Reply by fab6** [reply]
I expected to have a fixed time step size of e.g. `1e-3` s, but it did not work as the wrong solver was always updated. Now I know that before I adjust anything, I need to figure out how to compile it.

**Comment by fab6** [comment, marked as answer]
Ok, I figured out the cfl-based time stepping with
<img width="818" alt="image" src="https://user-images.githubusercontent.com/3047695/163252240-cc3a57cc-f307-450d-bcbe-5d4e0d54776b.png">
... a tiny step ;-)
